### PR TITLE
Fix issues with diagnostics emitted during manifest loading

### PR DESCRIPTION
**Pull request by hartbit**
Fixes two issues:

* Crash when a product references an unknown target in the root package
* Missing diagnostic when failing to load a manifest of a dependency package

### CODE REVIEW

File: Sources/PackageLoading/ManifestLoader.swift [hunk 1]
```diff
@@ -320,7 +320,7 @@ public final class ManifestLoader: ManifestLoaderProtocol {
             }
 
             // Check that products that reference only binary targets don't define a type.
-            let areTargetsBinary = product.targets.allSatisfy({ manifest.targetMap[$0]!.type == .binary })
+            let areTargetsBinary = product.targets.allSatisfy({ manifest.targetMap[$0]?.type == .binary })
```
Comment (aciidb0mb3r) [inline Sources/PackageLoading/ManifestLoader.swift]: Can you write a test for this?

Comment (hartbit) [reply Sources/PackageLoading/ManifestLoader.swift]: There is a test. Its in the commit.

Comment (hartbit) [reply Sources/PackageLoading/ManifestLoader.swift]: Its called `testProductTargetNotFound`. Without this change, that test crashes.

Comment (aciidb0mb3r) [reply Sources/PackageLoading/ManifestLoader.swift]: aah

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Pull request merged**